### DISCUSSION TITLE
How to reset the RadioGroup when used in a controlled manner?

### DISCUSSION THREAD

**Discussion by ckastbjerg** [radix-ui/primitives#1587] [Help] [answered]
Hi 👋 

I have a question about using [the `RadioGroup` component](https://www.radix-ui.com/docs/primitives/components/radio-group). 

I sort of need to use the component in a controlled manner. And I have a use case where I need to reset the radio group (so that none of its items are selected). The component only accepts a `value` of type `string` or `undefined`. And passing `undefined` doesn't seem to reset the radio group. Passing `null` on the other hand does reset it, but then causes a type error.

You can see an example of the issue here: https://codesandbox.io/s/heuristic-brook-x77cul?file=/App.js 

So my question is whether this is a bug or by design? Should `null` be allowed?

**Comment by benoitgrelard** [comment, marked as answer]
Hey @ckastbjerg, this looks like a bug potentially, similar to #1569.

**Reply by ckastbjerg** [reply]
@benoitgrelard thanks for the quick response 🙌 I've [reported this as a bug](https://github.com/radix-ui/primitives/issues/1588). Considering this answered.
